### ISSUE TITLE
Lack of Single, Maybe, Completable

### ISSUE THREAD

**Issue by ILAgent** [dotnet/reactive#1142]
Why don't we have these types? TPL Task is not the same thing and behaves otherwise.

**Comment by akarnokd** [comment]
It's .NET, you don't have to host everything in one project and still allow fluent interoperations. If you are such keen for them, they are available as 3rd party library: https://github.com/akarnokd/reactive-extensions#reactive-extensions

**Comment by ILAgent** [comment]
Any feature can be provided through a side library. But RxJava, RxJs and other Rx don't suggest me using a third-party bicycle. Rx.net looks like a son in law in the Reactive Streams family. Or maybe is there any more fundamental explanation?

**Comment by akarnokd** [comment]
Java doesn't have extension methods, therefore, the fluent API requires hosting all reactive sources in one library. I it had extension methods like C# or Kotlin, we would have split the library into smaller components years ago as many only use one or two reactive type in their project anyway.

**Comment by apobekiaris** [comment]
I agree with @ILAgent,  these operators have no dependencies physical or logical to justify splitting and although your @akarnokd lib is super cool it adds at least the support overhead of an extra dependency to our projects.

**Comment by ILAgent** [comment]
> Java doesn't have extension methods, therefore, the fluent API requires hosting all reactive sources in one library. If it had extension methods like C# or Kotlin, we would have split the library into smaller components years ago as many only use one or two reactive type in their project anyway.

The possibility to divide features by assemblies doesn't mean that they  shouldn't be implemented as a part of the framework. I don't want to search and use strange third-party libraries. I want rx.net to be feature-complete as same as RxJava.

**Comment by akarnokd** [comment]
Do you think I wouldn't be the most happy person to have Rx.NET be in par with RxJava regarding features? But I respect the other maintainers and the original goal of the library to not bloat it with all sorts of components just because. Also it is not just about adding code, I'd have to maintain them, document them, explain them and disseminate them to audiences who never even thought about flow cardinality.
